pkg/scheduling/requirement.go

```diff
@@ -187,6 +187,46 @@ func (r *Requirement) Intersection(requirement *Requirement) *Requirement {
 	return &Requirement{Key: r.Key, values: values, complement: complement, greaterThan: greaterThan, lessThan: lessThan, MinValues: minValues}
 }
 
+// nolint:gocyclo
+// HasIntersection is a more efficient implementation of Intersection
+// It validates whether there is an intersection between the two requirements without actually creating the sets
+// This prevents the garbage collector from having to spend cycles cleaning up all of these created set objects
+func (r *Requirement) HasIntersection(requirement *Requirement) bool {
+	greaterThan := maxIntPtr(r.greaterThan, requirement.greaterThan)
+	lessThan := minIntPtr(r.lessThan, requirement.lessThan)
+	if greaterThan != nil && lessThan != nil && *greaterThan >= *lessThan {
```

Comment:
Just double checking - is this supposed to be `*greaterThan >= *lessThan` and not `*greaterThan > *lessThan`? Wouldn't there be an intersection if these exist and have overlap?

Reply:
Nope, because you are saying that a requirement is valid if, for instance -- elem > 4 and also elem < 4 -- there is no such value that satisfies both